### ISSUE TITLE
Cannot read property 'name' of null on master.

### ISSUE THREAD

**Issue by ryantheleach** [bootstrap-tagsinput/bootstrap-tagsinput#604]
```
var classes = new Bloodhound({
    datumTokenizer: Bloodhound.tokenizers.obj.whitespace('text'),
    queryTokenizer: Bloodhound.tokenizers.whitespace,
    prefetch: 'classes.json'
});
classes.initialize();

var elt = $('.example_cms > > select');
elt.tagsinput({
    tagClass: function(item) {
        switch (item.continent) {
            case 'position' : return 'label label-primary';
            case 'padding'  : return 'label label-danger' ;
            case 'margin'   : return 'label label-success';
            case 'col-xs'   : return 'label label-default';
            case 'col-sm'   : return 'label label-info'   ;
            case 'col-md'   : return 'label label-warning';
            case 'col-lg'   : return 'label label-danger' ;
        }
    },
    itemValue: 'value',
    itemText: 'text',
    typeaheadjs: {
        name: 'classes',
        displayKey: 'text',
        source: classes.ttAdapter()
    },
    confirmKeys: [13, 38, 44]
});
```
classes.json
```
[ { "value": "position-flowing",  "text": "position-flowing",  "continent": "position"  },
  { "value": "position-relative", "text": "position-relative", "continent": "position"  },
  { "value": "position-absolute", "text": "position-absolute", "continent": "position"  },
  { "value": "padding-normal",    "text": "padding-normal",    "continent": "padding"   },
  { "value": "padding-more",      "text": "padding-more",      "continent": "padding"   },
  { "value": "padding-padded",    "text": "padding-padded",    "continent": "padding"   },
  { "value": "margin-normal",     "text": "margin-normal",     "continent": "margin"    },
  { "value": "margin-more",       "text": "margin-more",       "continent": "margin"    },
  { "value": "margin-padded",     "text": "margin-padded",     "continent": "margin"    },
  { "value": "col-xs-4",          "text": "col-xs-4",          "continent": "col-xs"    },
  { "value": "col-xs-6",          "text": "col-xs-6",          "continent": "col-xs"    },
  { "value": "col-xs-8",          "text": "col-xs-8",          "continent": "col-xs"    },
  { "value": "col-xs-12",         "text": "col-xs-12",         "continent": "col-xs"    },
  { "value": "col-sm-4",          "text": "col-sm-4",          "continent": "col-sm"    },
  { "value": "col-sm-6",          "text": "col-sm-6",          "continent": "col-sm"    },
  { "value": "col-sm-8",          "text": "col-sm-8",          "continent": "col-sm"    },
  { "value": "col-sm-12",         "text": "col-sm-12",         "continent": "col-sm"    },
  { "value": "col-md-4",          "text": "col-md-4",          "continent": "col-md"    },
  { "value": "col-md-6",          "text": "col-md-6",          "continent": "col-md"    },
  { "value": "col-md-8",          "text": "col-md-8",          "continent": "col-md"    },
  { "value": "col-md-12",         "text": "col-md-12",         "continent": "col-md"    },
  { "value": "col-lg-4",          "text": "col-lg-4",          "continent": "col-lg"    },
  { "value": "col-lg-6",          "text": "col-lg-6",          "continent": "col-lg"    },
  { "value": "col-lg-8",          "text": "col-lg-8",          "continent": "col-lg"    },
  { "value": "col-lg-12",         "text": "col-lg-12",         "continent": "col-lg"    }
]

```

Index.html (edited into example Index.html)

```

        <div class="example example_cms">
            <h3>CMS</h3>
            <div class="bs-example">
                <select multiple data-role="tagsinput">
                    <option value="position-flowing">position-flowing, positioned according to the normal flow of the page</option>
                    <option value="position-relative">position-relative, positioned relative to its normal position</option>
                    <option value="position-absolute">position-absolute, positioned relative to the outer form area (and parents)</option>
                    <option value="padding-normal">padding-normal</option>
                    <option value="padding-more">padding-more</option>
                    <option value="padding-padded">padding-padded</option>
                    <option value="margin-normal">margin-normal</option>
                    <option value="margin-more">margin-more</option>
                    <option value="margin-padded">margin-padded</option>
                    <option value="col-xs-4">col-xs-4</option>
                    <option value="col-xs-6">col-xs-6</option>
                    <option value="col-xs-8">col-xs-8</option>
                    <option value="col-xs-12">col-xs-12</option>
                    <option value="col-sm-4">col-sm-4</option>
                    <option value="col-sm-6">col-sm-6</option>
                    <option value="col-sm-8">col-sm-8</option>
                    <option value="col-sm-12">col-sm-12</option>
                    <option value="col-md-4">col-md-4</option>
                    <option value="col-md-6">col-md-6</option>
                    <option value="col-md-8">col-md-8</option>
                    <option value="col-md-12">col-md-12</option>
                    <option value="col-lg-4">col-lg-4</option>
                    <option value="col-lg-6">col-lg-6</option>
                    <option value="col-lg-8">col-lg-8</option>
                    <option value="col-lg-12">col-lg-12</option>
                </select>
            </div>
            <div class="accordion">
                <div class="accordion-group">
                    <div class="accordion-heading">
                        <a class="accordion-toggle" data-toggle="collapse" href="#accordion_example_markup">
                            Show code
                        </a>
                    </div>
                    <div id="accordion_example_cms" class="accordion-body collapse">
                        <div class="accordion-inner highlight">
                            <pre><code data-language="html">&lt;input type=&quot;text&quot; value=&quot;Amsterdam,Washington,Sydney,Beijing,Cairo&quot; data-role=&quot;tagsinput&quot; /&gt;</code></pre>
                        </div>
                    </div>
                </div>
            </div>
            <table class="table table-bordered table-condensed"><thead><tr><th>statement</th><th>returns</th></tr></thead><tbody><tr><td><code>$("input").val()</code></td><td><pre class="val"><code data-language="javascript"></code></pre></td></tr><tr><td><code>$("input").tagsinput('items')</code></td><td><pre class="items"><code data-language="javascript"></code></pre></td></tr></tbody></table>
        </div>
```

Results in a console error of 

bootstrap-tagsinput.js:362 Uncaught TypeError: Cannot read property 'name' of null
    at bootstrap-tagsinput.js:362
    at Array.some (<anonymous>)
    at TagsInput.<anonymous> (bootstrap-tagsinput.js:361)
    at HTMLInputElement.f (jquery.min.js:2)
    at HTMLInputElement.dispatch (jquery.min.js:3)
    at HTMLInputElement.r.handle (jquery.min.js:3)
    at Object.trigger (jquery.min.js:3)
    at HTMLInputElement.<anonymous> (jquery.min.js:3)
    at Function.each (jquery.min.js:2)
    at n.fn.init.each (jquery.min.js:2)

**Comment by yingliangzhang** [comment]
Same issue here

**Comment by Littlericket** [comment]
Had the same issue, fixed it by downloading typeahead 1.2.0 from https://github.com/corejavascript/typeahead.js/releases, and changing line 372 - 378 in tagsinput.js to

```
$.fn.typeahead.apply(self.$input, typeaheadjs).on('typeahead:selected', $.proxy(function (obj, datum) {
  // @TODO Dep: https://github.com/corejavascript/typeahead.js/issues/89
  self.add(datum);
  
  self.$input.typeahead('val', '');
}, self));
```

You need to make the bloodhound return a object and define "itemValue" and "itemText" in tagsinput for it to work.

**Comment by Sam-Butler** [comment]
Same issue, @Littlericket 's suggestion worked with the latest typeahead and slightly different line numbers in tagsinput.js. The object returned was 
`{
  id: ...,
  name: ...
}`

So I defined itemValue and itemText as 'id' in the tagsinput initialisation, which worked as expected.

**Comment by alhaji-aki** [comment]
I am still getting this error. I am kind of confused where to place @Littlericket's solution

**Comment by ryantheleach** [comment]
I have unsubscribed as I currently have no involvement with the project I first encountered this in.

I will leave it open, but do not expect me to reply again, or 'own' this ticket.

**Comment by zonair-r** [comment]
> Had the same issue, fixed it by downloading typeahead 1.2.0 from https://github.com/corejavascript/typeahead.js/releases, and changing line 372 - 378 in tagsinput.js to
> 
> ```
> $.fn.typeahead.apply(self.$input, typeaheadjs).on('typeahead:selected', $.proxy(function (obj, datum) {
>   // @TODO Dep: https://github.com/corejavascript/typeahead.js/issues/89
>   self.add(datum);
>   
>   self.$input.typeahead('val', '');
> }, self));
> ```
> 
> You need to make the bloodhound return a object and define "itemValue" and "itemText" in tagsinput for it to work.

Unable to get ahead of this error, the tags input initialization is
	var countries = new Bloodhound({
	  datumTokenizer: Bloodhound.tokenizers.obj.whitespace('name'),
	  queryTokenizer: Bloodhound.tokenizers.whitespace,
	  prefetch: {
		url: 'data/countries.json',
		filter: function(list) {
		  return $.map(list, function(name) {
			return { name: name }; });
		}
	  }
	});
	countries.initialize();

	$('#tags-input').tagsinput({
	  typeaheadjs: {
		name: 'countries',
		displayKey: 'name',
		valueKey: 'name',
		source: countries.ttAdapter()

	},
});

And Made the change, as mentioned by @Littlericket . Help much appreciated

**Comment by micktaiwan** [comment]
Does this pull request fix this error ? https://github.com/bootstrap-tagsinput/bootstrap-tagsinput/pull/620

**Comment by vuduchong209305** [comment]
> Had the same issue, fixed it by downloading typeahead 1.2.0 from https://github.com/corejavascript/typeahead.js/releases, and changing line 372 - 378 in tagsinput.js to
> 
> ```
> $.fn.typeahead.apply(self.$input, typeaheadjs).on('typeahead:selected', $.proxy(function (obj, datum) {
>   // @TODO Dep: https://github.com/corejavascript/typeahead.js/issues/89
>   self.add(datum);
>   
>   self.$input.typeahead('val', '');
> }, self));
> ```
> 
> You need to make the bloodhound return a object and define "itemValue" and "itemText" in tagsinput for it to work.

thank.
